File: Programmers/연습문제/Level 1/정수 제곱근 판별/Solution.js
function solution(n) {
    const root = getRoot(n);
    const answer = getAnswer(n, root);
    return answer;
}

function getRoot(n) {
    return parseInt(Math.sqrt(n));
}

function getAnswer(n, root) {
    return root ** 2 == n ? (root + 1) ** 2 : -1;
}

File: Programmers/연습문제/Level 1/정수 제곱근 판별/Solution.swift
import Foundation

func solution(_ n:Int64) -> Int64 {
    let root = Int64(sqrt(Double(n)))
    return root * root == n ? (root + 1) * (root + 1) : -1
}


// TC
let n: Int64 = 121
print(solution(n)) // 144
